### PR TITLE
[DOC] RDoc for Open3

### PR DESCRIPTION
**Pull request by BurdetteLamar**
@peterzhu2118, a question:  are the locally-used hash entries in +options+ removed?

Also, I'll add Related: later (after I grok the other capture*).

### CODE REVIEW

File: lib/open3.rb [hunk 1]
```diff
@@ -482,44 +482,95 @@ class << self
...
-  # <code>opts[:stdin_data]</code> and <code>opts[:binmode]</code>.  See Process.spawn.
+  # - Creates a child process, by calling Open3.popen3 with the given arguments
+  #   (except for certain entries in hash +options+; see below).
+  # - Returns as strings +stdout+ and +stderr+ the standard output
+  #   and standard error of the child process.
+  # - Returns as +status+ a <tt>Process::Status</tt> object
+  #   that represents the exit status of the child process.
   #
-  # If <code>opts[:stdin_data]</code> is specified, it is sent to the command's standard input.
+  # With no block given, returns the array <tt>[stdin_s, stdout_a, status]</tt>:
```
Comment (peterzhu2118) [inline lib/open3.rb:498]: I think this is confusing because it does not accept a block.

File: lib/open3.rb [hunk 2]
```diff
@@ -482,44 +482,95 @@ class << self
...
+  # Argument +options+ is a hash of options for the new process;
+  # see {Execution Options}[rdoc-ref:Process@Execution+Options].
+  #
+  # The hash +options+ is passed to method Open3.popen3;
+  # two options have local effect in method Open3.capture3:
+  #
+  # - If entry <tt>options[:stdin_data]</tt> exists,
+  #   and its string value is sent to the command's standard input:
+  #
+  #     Open3.capture3('tee', {:stdin_data => 'Foo'})
```
Comment (peterzhu2118) [inline lib/open3.rb:519]: Nit: a hash is not needed for keyword arguments.

```suggestion
  #     Open3.capture3('tee', stdin_data: 'Foo')
```

**Comment by peterzhu2118** [comment]
> are the locally-used hash entries in +options+ removed?

It does look like the `stdin_data` and `binmode` keys are removed from `options` before it is passed into `popen3`.

File: lib/open3.rb (hunk 1 above)
Comment (BurdetteLamar) [reply lib/open3.rb]: Fixed.

File: lib/open3.rb (hunk 2 above)
Comment (BurdetteLamar) [reply lib/open3.rb]: Fixed.

File: lib/open3.rb [hunk 3]
```diff
@@ -482,44 +482,95 @@ class << self
...
+  # Argument +options+ is a hash of options for the new process;
+  # see {Execution Options}[rdoc-ref:Process@Execution+Options].
+  #
+  # The hash +options+ is passed to method Open3.popen3;
+  # two options have local effect in method Open3.capture3:
+  #
+  # - If entry <tt>options[:stdin_data]</tt> exists, the entry is removed
+  #   and its string value is sent to the command's standard input:
+  #
+  #     Open3.capture3('tee', :stdin_data => 'Foo')
```
Comment (peterzhu2118) [inline lib/open3.rb:519]: I'm curious why you're using the old style of hashes literals with symbol keys instead of the new style `stdin_data: 'Foo'`?

Comment (BurdetteLamar) [reply lib/open3.rb]: Fixed.  Actually, I much prefer the new notation;  don't know why I did that.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**